File: android/lca/src/main/java/com/dreampany/lca/data/source/room/RoomGraphDataSource.java
/*
package com.dreampany.lca.data.source.room;

import com.dreampany.lca.data.misc.GraphMapper;
import com.dreampany.lca.data.model.Graph;
import com.dreampany.lca.data.source.api.GraphDataSource;

import java.util.List;

import javax.inject.Singleton;

import com.dreampany.lca.data.source.dao.GraphDao;
import io.reactivex.Maybe;

*/
/**
 * Created by Hawladar Roman on 30/5/18.
 * Dreampany Ltd
 * dreampanymail@gmail.com
 *//*

@Singleton
public class RoomGraphDataSource implements GraphDataSource {

    private final GraphMapper mapper;
    private final GraphDao dao;

    public RoomGraphDataSource(GraphMapper mapper,
                               GraphDao dao) {
        this.mapper = mapper;
        this.dao = dao;
    }

    @Override
    public boolean isEmpty() {
        return getCount() == 0;
    }

    @Override
    public Maybe<Boolean> isEmptyRx() {
        return Maybe.fromCallable(this::isEmpty);
    }

    @Override
    public int getCount() {
        return dao.getCount();
    }

    @Override
    public Maybe<Integer> getCountRx() {
        return null;
    }

    @Override
    public boolean isExists(Graph graph) {
        return false;
    }

    @Override
    public Maybe<Boolean> isExistsRx(Graph graph) {
        return null;
    }

    @Override
    public long putItem(Graph graph) {
        return 0;
    }

    @Override
    public Maybe<Long> putItemRx(Graph graph) {
        return Maybe.fromCallable(() -> putItem(graph));
    }

    @Override
    public List<Long> putItems(List<Graph> graphs) {
        return null;
    }

    @Override
    public Maybe<List<Long>> putItemsRx(List<Graph> graphs) {
        return null;
    }

    @Override
    public int delete(Graph graph) {
        return 0;
    }

    @Override
    public Maybe<Integer> deleteRx(Graph graph) {
        return null;
    }

    @Override
    public List<Long> delete(List<Graph> graphs) {
        return null;
    }

    @Override
    public Maybe<List<Long>> deleteRx(List<Graph> graphs) {
        return null;
    }

    @Override
    public Graph getItem(String id) {
        return null;
    }

    @Override
    public Maybe<Graph> getItemRx(String id) {
        return null;
    }

    @Override
    public List<Graph> getItems() {
        return null;
    }

    @Override
    public Maybe<List<Graph>> getItemsRx() {
        return null;
    }

    @Override
    public List<Graph> getItems(int limit) {
        return null;
    }

    @Override
    public Maybe<List<Graph>> getItemsRx(int limit) {
        return null;
    }

    @Override
    public Graph getItem(String slug, long startTime, long endTime) {
        return null;
    }

    @Override
    public Maybe<Graph> getItemRx(String slug, long startTime, long endTime) {
        return dao.getItemRx(slug, startTime, endTime);
    }
}
*/

File: android/lca/src/main/kotlin/com/dreampany/lca/data/source/room/RoomGraphDataSource.kt
package com.dreampany.lca.data.source.room

import com.dreampany.lca.data.misc.GraphMapper
import com.dreampany.lca.data.model.Graph
import com.dreampany.lca.data.source.api.GraphDataSource
import com.dreampany.lca.data.source.dao.GraphDao
import io.reactivex.Maybe
import javax.inject.Singleton

/**
 * Created by roman on 2019-07-14
 * Copyright (c) 2019 bjit. All rights reserved.
 * hawladar.roman@bjitgroup.com
 * Last modified $file.lastModified
 */
@Singleton
class RoomGraphDataSource constructor(
    val mapper: GraphMapper,
    val dao: GraphDao
) : GraphDataSource {

    override fun isEmpty(): Boolean {
        return getCount() == 0
    }

    override fun isEmptyRx(): Maybe<Boolean> {
        return Maybe.fromCallable( { this.isEmpty() })
    }

    override fun getCount(): Int {
        return dao.count
    }

    override fun getCountRx(): Maybe<Int> {
        return Maybe.empty()
    }

    override fun isExists(graph: Graph): Boolean {
        return false
    }

    override fun isExistsRx(graph: Graph): Maybe<Boolean> {
        return Maybe.empty()
    }

    override fun putItem(graph: Graph): Long {
        return 0
    }

    override fun putItemRx(graph: Graph): Maybe<Long> {
        return Maybe.fromCallable { putItem(graph) }
    }

    override  fun putItems(graphs: List<Graph>): List<Long>? {
        return null
    }

    override  fun putItemsRx(graphs: List<Graph>): Maybe<List<Long>> {
        return Maybe.empty()
    }

    override fun delete(graph: Graph): Int {
        return 0
    }

    override fun deleteRx(graph: Graph): Maybe<Int> {
        return Maybe.empty()
    }

    override fun delete(graphs: List<Graph>): List<Long>? {
        return null
    }

    override fun deleteRx(graphs: List<Graph>): Maybe<List<Long>> {
        return Maybe.empty()
    }

    override fun getItem(id: String): Graph? {
        return null
    }

    override fun getItemRx(id: String): Maybe<Graph> {
        return Maybe.empty()
    }

    override fun getItems(): List<Graph>? {
        return null
    }

    override fun getItemsRx(): Maybe<List<Graph>> {
        return Maybe.empty()
    }

    override fun getItems(limit: Int): List<Graph>? {
        return null
    }

    override fun getItemsRx(limit: Int): Maybe<List<Graph>> {
        return Maybe.empty()
    }

    override fun getItem(slug: String, startTime: Long, endTime: Long): Graph? {
        return null
    }

    override fun getItemRx(slug: String, startTime: Long, endTime: Long): Maybe<Graph> {
        return dao.getItemRx(slug, startTime, endTime)
    }
}